extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/S3DataSegmentPuller.java

```diff
@@ -231,6 +229,8 @@ public Writer openWriter()
       @Override
       public long getLastModified()
       {
+        final S3ObjectSummary objectSummary =
```

Comment:
Making the LIST call here every time `getLastModified` is called seems costly.
Maybe initialize it lazily but only once the way `s3Object` is being initialized in `openInputStream`?

Reply:
You could also check if `s3Object` is already initialized and just use `s3Object.getObjectMetadata().getLastModified()`.
If not initialized, then you could go for the LIST (or GET if it is cheaper).